urllib3/util/connection.py

```diff
@@ -26,22 +23,13 @@ def is_connection_dropped(conn):  # Platform-specific
     if sock is None:  # Connection already closed (such as by httplib).
         return True
 
-    if not poll:
-        if not select:  # Platform-specific: AppEngine
-            return False
-
-        try:
-            return select([sock], [], [], 0.0)[0]
-        except socket.error:
-            return True
+    if not HAS_SELECT:
+        return False
 
-    # This version is better on platforms that support it.
-    p = poll()
-    p.register(sock, POLLIN)
-    for (fno, ev) in p.poll(0.0):
-        if fno == sock.fileno():
-            # Either data is buffered (bad), or the connection is dropped.
-            return True
+    try:
+        return wait_for_read(sock, timeout=0.0)
+    except select.error:
```

Comment:
It might be nicer to provide a custom exception we can track, rather than piggybacking on the stdlib here.